impl/doc/into.md

```diff
@@ -1,166 +1,119 @@
-struct MyInt8(i8);
-
-// Even for ref types
-#[derive(Into, PartialEq)]
-#[into(owned, ref(types(i64)))]
-struct MyInt64Wrapped(MyInt64);
-
-assert!(i32::from(MyInt(2)) == 2i32);
-assert!(i64::from(MyInt64(6)) == 6i64);
-assert!(<&i64>::from(&MyInt64(6)) == &6i64);
-assert!(<&mut i64>::from(&mut MyInt64(6)) == &mut 6i64);
-assert!(i8::from(MyInt8(7)) == 7i8);
-assert!(i16::from(MyInt8(7)) == 7i16);
-assert!(i32::from(MyInt8(7)) == 7i32);
-assert!(MyInt64::from(MyInt64Wrapped(MyInt64(1))) == MyInt64(1));
-assert!(<&MyInt64>::from(&MyInt64Wrapped(MyInt64(1))) == &MyInt64(1));
-assert!(<&i64>::from(&MyInt64Wrapped(MyInt64(1))) == &1i64);
-```
-
-
-
+#[derive(Debug, Into, PartialEq)]
+struct Int(i32);
 
-## Tuple structs
+assert_eq!(2, Int(2).into());
+```
 
-When deriving `Into` for a tuple struct with a single field (i.e. a newtype) like this:
+For structs that have multiple fields `.into()` extract tuple containing the
+desired content for each field.
 
 ```rust
 # use derive_more::Into;
 #
-#[derive(Into)]
-struct MyInt(i32);
-```
-
-Code like this will be generated:
+#[derive(Debug, Into, PartialEq)]
+struct Point(i32, i32);
 
-```rust
-# struct MyInt(i32);
-impl ::core::convert::From<MyInt> for (i32) {
-    fn from(original: MyInt) -> (i32) {
-        (original.0)
-    }
-}
+assert_eq!((1, 2), Point(1, 2).into());
 ```
 
-The behaviour is a bit different when deriving for a struct with multiple
-fields, since it returns a tuple. For instance when deriving for a tuple struct
-with two fields like this:
+To specify concrete types to derive convert into use `#[into(<types>)]`.
 
 ```rust
+# use std::borrow::Cow;
+#
 # use derive_more::Into;
 #
-#[derive(Into)]
-struct MyInts(i32, i32);
-```
+#[derive(Debug, Into, PartialEq)]
+#[into(Cow<'static, str>, String)]
+struct Str(Cow<'static, str>);
 
-Code like this will be generated:
+assert_eq!("String".to_owned(), String::from(Str("String".into())));
+assert_eq!(Cow::Borrowed("Cow"), <Cow<_>>::from(Str("Cow".into())));
 
-```rust
-# struct MyInts(i32, i32);
-impl ::core::convert::From<MyInts> for (i32, i32) {
-    fn from(original: MyInts) -> (i32, i32) {
-        (original.0, original.1)
-    }
+#[derive(Debug, Into, PartialEq)]
+#[into((i64, i64), (i32, i32))]
+struct Point {
+    x: i32,
+    y: i32,
 }
-```
-
-
 
+assert_eq!((1_i64, 2_i64), Point { x: 1_i32, y: 2_i32 }.into());
+assert_eq!((3_i32, 4_i32), Point { x: 3_i32, y: 4_i32 }.into());
+```
 
-## Regular structs
-
-For regular structs almost the same code is generated as for tuple structs
-except in the way the field values are assigned to the new struct.
-When deriving for a regular struct with a single field like this:
+In addition to converting to owned types, this macro supports deriving into
+reference, mutable or not via `#[into(ref(...))]`/`#[into(ref_mut(...))]`.
 
 ```rust
 # use derive_more::Into;
 #
-#[derive(Into)]
-struct Point1D {
-    x: i32,
-}
-```
+#[derive(Debug, Into, PartialEq)]
+#[into(owned, ref(i32), ref_mut)]
```

Comment:
So `into(owned)` is equivalent to `into(types({current_type}))`?

Reply:
I think a better example would be to use `ref(i64)`

Reply:
hmm, actually what's the difference between `ref(i64)` and simply using `&i64`?

Reply:
@JelteF here `ref`/`ref_mut` control `Self` type representation, while `&i64` controls target type.

`ref(i64)` means `impl Into<&i64> for &MyType`, while `&i64` means only `impl Into<&i64> for MyType`.